bootstraptest/test_yjit.rb

```diff
@@ -3690,3 +3690,45 @@ def literal_append(sql, v)
   )
   foo 1
 }
+
+# Regression test for CantCompile not using starting_ctx
+assert_normal_exit %q{
+  class Integer
+    def ===(other)
+      false
+    end
+  end
+
+  def my_func(x)
+    case x
+    when 1
+      1
+    when 2
+      2
+    else
+      3
+    end
+  end
+
+  my_func(1)
```

Comment:
Do you remember how this test failed when you didn't use `starting_ctx`? I assume this is about `CantCompile` on `opt_case_dispatch` due to `Integer#===` redefinition, but it doesn't seem to push/pop stack operands before that `CantCompile`.

Reply:
well, sorry, never mind. I was on a branch that removed `starting_ctx` from `opt_case_dispatch` 🙂 That's why.